opencog/query/PatternUtils.cc

```diff
@@ -117,21 +116,18 @@ void get_connected_components(
 			size_t nc = components.size();
 			for (size_t i = 0; i<nc; i++)
 			{
-				std::set<Handle> cur_vars(component_vars[i]);
```

Comment:
I can add the constructor `Handle(std::shared_ptr<const Atom> atom);` but I don't understand the problem that its solving.  I really would strongly prefer not to introduce ConstHandle.   Can you show some assembly code generated by gcc-4.8 where some sort of register reload could actually be avoided by using ConstHandle?